crates/wit-component/tests/interfaces.rs

```diff
@@ -54,43 +54,37 @@ fn run_test(path: &Path, is_dir: bool) -> Result<()> {
         resolve.push_file(path)?
     };
 
-    for package in packages {
-        assert_print(&resolve, &[package], path, is_dir)?;
+    assert_print(&resolve, &packages, path, is_dir)?;
 
-        let features = WasmFeatures::default() | WasmFeatures::COMPONENT_MODEL;
+    let features = WasmFeatures::default() | WasmFeatures::COMPONENT_MODEL;
 
-        // First convert the WIT package to a binary WebAssembly output, then
-        // convert that binary wasm to textual wasm, then assert it matches the
-        // expectation.
-        let wasm = wit_component::encode(Some(true), &resolve, package)?;
-        let wat = wasmprinter::print_bytes(&wasm)?;
-        assert_output(&path.with_extension("wat"), &wat)?;
-        wasmparser::Validator::new_with_features(features)
-            .validate_all(&wasm)
-            .context("failed to validate wasm output")?;
+    // First convert the WIT package to a binary WebAssembly output, then
+    // convert that binary wasm to textual wasm, then assert it matches the
+    // expectation.
+    let wasm = wit_component::encode(Some(true), &resolve, &packages)?;
+    let wat = wasmprinter::print_bytes(&wasm)?;
+    assert_output(&path.with_extension("wat"), &wat)?;
+    wasmparser::Validator::new_with_features(features)
+        .validate_all(&wasm)
+        .context("failed to validate wasm output")?;
 
-        // Next decode a fresh WIT package from the WebAssembly generated. Print
-        // this package's documents and assert they all match the expectations.
-        let decoded = wit_component::decode(&wasm)?;
-        assert_eq!(
-            1,
-            decoded.packages().len(),
-            "Each input WIT package should produce WASM that contains only one package"
-        );
+    // Next decode a fresh WIT package from the WebAssembly generated. Print
+    // this package's documents and assert they all match the expectations.
+    let decoded = wit_component::decode(&wasm)?;
 
-        let decoded_package = decoded.packages()[0];
-        let resolve = decoded.resolve();
+    let decoded_package = decoded.packages()[0];
+    let resolve = decoded.resolve();
 
-        assert_print(resolve, decoded.packages(), path, is_dir)?;
+    assert_print(resolve, decoded.packages(), path, is_dir)?;
 
-        // Finally convert the decoded package to wasm again and make sure it
-        // matches the prior wasm.
-        let wasm2 = wit_component::encode(Some(true), resolve, decoded_package)?;
-        if wasm != wasm2 {
-            let wat2 = wasmprinter::print_bytes(&wasm)?;
-            assert_eq!(wat, wat2, "document did not roundtrip correctly");
-        }
+    // Finally convert the decoded package to wasm again and make sure it
+    // matches the prior wasm.
+    let wasm2 = wit_component::encode(Some(true), resolve, &[decoded_package])?;
```

Comment:
I'm a bit surprised this test is passing because I would expect that the `decoded_package` argument should be the result of `decoded.packages()` now that multiple packages are supported. Could you investigate if a new test is needed to reflect that?

Reply:
Looks like the assertion was wrong!  accidentally setting `wat2` to be `wasm` instead of `wasm2` when checking the diff.  You're right, when fixing the assertion, the test fails as is, and changing `decoded_package` to meet your expectations fixes it.

Reply:
Turns out when I fixed it fixed the assertion, it came out that the docs are being stored in the wat in a different order depending on if a wit is being parsed vs a binary, in such a way that it doesn't impact how the docs get printed I believe.  Can fix the tests by performing a sort on the metadata fields, but still investigating how to just get encode things in the same order regardless.

Reply:
Is that a regression from this PR? Or a preexisting issue?

Reply:
I think this is a preexisting issue.

```rs
if wasm != wasm2 {
        let wat2 = wasmprinter::print_bytes(&wasm)?;
        assert_eq!(wat, wat2, "document did not roundtrip correctly");
}
```
If I understand correctly, `assert_eq` is receiving two different `wat` samples created from `wasm`, neither represent `wasm2`...  Was spending some time trying to dig into this, and it looks like modifying main also has this issue of doc metadata being reordered if I'm reading the code correctly...